File: mvvm-test/src/main/java/com/martin/mvvm/db/dao/UserDao.java
package com.martin.mvvm.db.dao;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import com.martin.mvvm.db.bean.User;

import java.util.List;

@Dao
public interface UserDao {

    @Query("SELECT * FROM user")
    LiveData<List<User>> getAllUsers();

    @Update
    void updateUser(User user);

    @Insert
    void insertUser(User user);
}

File: room-livedata-recyclerview/src/main/java/com/example/rlr/UserDao.kt
package com.example.rlr

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.Query

@Dao
interface UserDao {
    @Query("SELECT * FROM users1 ORDER BY name ASC")
    fun getAllUsers(): LiveData<List<User?>?>? // 返回 LiveData

    @Insert
    fun insert(user: User) : Long

    @Delete
    fun delete(user: User?)
}
